components/core/src/clp/CurlDownloadHandler.cpp

```diff
@@ -56,6 +60,60 @@ CurlDownloadHandler::CurlDownloadHandler(
         m_http_headers.append("Cache-Control: no-cache");
         m_http_headers.append("Pragma: no-cache");
     }
+    if (http_header_kv_pairs.has_value()) {
+        constexpr std::array<std::string_view, 3> cReservedHeaders
+                = {"range", "cache-control", "pragma"};
+        // RFC 7230 token pattern: one or more tchars
+        std::regex const header_name_pattern("^(?![!#$%&'*+.^_`|~-])[!#$%&'*+.^_`|~0-9a-zA-Z-]+$");
+        // Must consist of printable ASCII characters (values between 0x20 and 0x7E)
+        std::regex const header_value_pattern("^[\\x20-\\x7E]*$");
+        for (auto const& [key, value] : http_header_kv_pairs.value()) {
+            // Convert to lowercase for case-insensitive comparison
+            std::string lower_key = key;
+            std::transform(
+                    lower_key.begin(),
+                    lower_key.end(),
+                    lower_key.begin(),
+                    [](unsigned char c) { return std::tolower(c); }
+            );
+
+            if (cReservedHeaders.end()
+                == std::find(cReservedHeaders.begin(), cReservedHeaders.end(), lower_key))
+            {
+                // Filter out illegal header names and header values by regex
+                if (false == value.empty() && ('\r' != value.back() && '\n' != value.back())) {
```

Comment:
CRLF should be two characters: `\r\n`. Maybe use [ends_with](https://en.cppreference.com/w/cpp/string/basic_string/ends_with)?